### ISSUE TITLE
UNREACHABLE executed at lib/BitcastUtils.cpp:1218

### ISSUE THREAD

**Issue by zwang20** [google/clspv#1204]
Compiling the attached source with the following command line

```
unzip a.zip
clspv -x ir -cl-mad-enable -DDEVICE_INFO=32 -D__SIZEOF_HOST_SIZE_T__=8 -DDEV_VER_MAJOR=3 -DDEV_VER_MINOR=0 -D_OPENCL_COMPILER -DPLAINTEXT_LENGTH=19 -DUCS_2 -DCONST_CACHE_SIZE=65536 -DASCII -DASCII -DLOC_0=-1 -DLOC_1=-1 -DLOC_2=-1 -DLOC_3=-1 -DNUM_INT_KEYS=1 -DIS_STATIC_GPU_MASK=0 -cl-single-precision-constant -cl-kernel-arg-info -rounding-mode-rte=16,32,64 -rewrite-packed-structs -std430-ubo-layout -decorate-nonuniform -arch=spir --use-native-builtins=acos,acosh,acospi,asin,asinh,asinpi,atan,atan2,atan2pi,atanh,atanpi,ceil,copysign,fabs,fdim,floor,fma,fmax,fmin,frexp,half_rsqrt,half_sqrt,isequal,isfinite,isgreater,isgreaterequal,isinf,isless,islessequal,islessgreater,isnan,isnormal,isnotequal,isordered,isunordered,ldexp,mad,rint,round,rsqrt,signbit,sqrt,tanh,trunc, -spv-version=1.6 -max-pushconstant-size=256 -max-ubo-size=65536 -global-offset -long-vector -module-constants-in-storage-buffer -cl-arm-non-uniform-work-group-size -enable-printf -printf-buffer-size=1048576 a.bc
```

results in the following output

```
Err: SrcTy = i32 - DstTy = i8 - Ty = i32 - CstVal = 32
Unexpected offset for type in GetIdxsForTyFromOffset
UNREACHABLE executed at /path/to/clvk/external/clspv/lib/BitcastUtils.cpp:1218!
```

[a.zip](https://github.com/google/clspv/files/12450111/a.zip)

**Comment by zwang20** [comment]
This particular example comes from `john --test --format=oldoffice-opencl`

**Comment by zwang20** [comment]
This bug appears to be fixed as of 619ad2d233afe0ff04b9162d46c32b4ed00e8fd2
